src/EFCore.PG/Storage/Internal/Mapping/NpgsqlNetworkTypeMappings.cs

```diff
@@ -30,58 +30,118 @@
 {
+    /// <summary>
+    /// The type mapping for the PostgreSQL macaddr type.
+    /// </summary>
+    /// <remarks>
+    /// See: https://www.postgresql.org/docs/current/static/datatype-net-types.html#DATATYPE-MACADDR
+    /// </remarks>
     public class NpgsqlMacaddrTypeMapping : NpgsqlTypeMapping
     {
+        /// <summary>
+        /// Constructs an instance of the <see cref="NpgsqlMacaddrTypeMapping"/> class.
+        /// </summary>
         public NpgsqlMacaddrTypeMapping() : base("macaddr", typeof(PhysicalAddress), NpgsqlDbType.MacAddr) {}
 
+        /// <summary>
+        /// Constructs an instance of the <see cref="NpgsqlMacaddrTypeMapping"/> class.
+        /// </summary>
+        /// <param name="parameters">The parameters for this mapping.</param>
         protected NpgsqlMacaddrTypeMapping(RelationalTypeMappingParameters parameters)
             : base(parameters, NpgsqlDbType.MacAddr) {}
 
+        /// <inheritdoc />
         protected override RelationalTypeMapping Clone(RelationalTypeMappingParameters parameters)
             => new NpgsqlMacaddrTypeMapping(parameters);
 
+        /// <inheritdoc />
         protected override string GenerateNonNullSqlLiteral(object value)
             => $"MACADDR '{(PhysicalAddress)value}'";
     }
 
+    /// <summary>
+    /// The type mapping for the PostgreSQL macaddr8 type.
+    /// </summary>
+    /// <remarks>
+    /// See: https://www.postgresql.org/docs/current/static/datatype-net-types.html#DATATYPE-MACADDR8
+    /// </remarks>
     public class NpgsqlMacaddr8TypeMapping : NpgsqlTypeMapping
     {
+        /// <summary>
+        /// Constructs an instance of the <see cref="NpgsqlMacaddr8TypeMapping"/> class.
+        /// </summary>
         public NpgsqlMacaddr8TypeMapping() : base("macaddr8", typeof(PhysicalAddress), NpgsqlDbType.MacAddr8) {}
 
+        /// <summary>
+        /// Constructs an instance of the <see cref="NpgsqlMacaddr8TypeMapping"/> class.
+        /// </summary>
+        /// <param name="parameters">The parameters for this mapping.</param>
         protected NpgsqlMacaddr8TypeMapping(RelationalTypeMappingParameters parameters)
             : base(parameters, NpgsqlDbType.MacAddr8) {}
 
+        /// <inheritdoc />
         protected override RelationalTypeMapping Clone(RelationalTypeMappingParameters parameters)
             => new NpgsqlMacaddr8TypeMapping(parameters);
 
+        /// <inheritdoc />
         protected override string GenerateNonNullSqlLiteral(object value)
             => $"MACADDR8 '{(PhysicalAddress)value}'";
     }
 
+    /// <summary>
+    /// The type mapping for the PostgreSQL inet type.
+    /// </summary>
+    /// <remarks>
+    /// See: https://www.postgresql.org/docs/current/static/datatype-net-types.html#DATATYPE-INET
+    /// </remarks>
     public class NpgsqlInetTypeMapping : NpgsqlTypeMapping
     {
+        /// <summary>
+        /// Constructs an instance of the <see cref="NpgsqlInetTypeMapping"/> class.
```

Comment:
I know that it's sometimes common practice to have these, but I really don't see the point of documentation on a constructor that says that it constructs the class...

Reply:
Long term, I would like to see us enable the compiler warning for undocumented public APIs. Also, FWIW, it seems to be [followed](https://docs.microsoft.com/en-us/ef/core/api/microsoft.entityframeworkcore.dbupdateexception#constructors) by the EF Core team too.

Reply:
There are a lot of such documentation in .NET, as @austindrenski said. Personally I think that for publics it should be everywhere. It's possible to read the source code since Npgsql is open source, but from the consumer's point it's not a good idea to jump to the code and read it instead of a comment.

Reply:
If both of you prefer it like this I'll defer, but once again IMHO this specific kind of documentation has zero added value for anyone - it repeats information that is already self-evident (constructors construct things). Nobody will learn something new from it that they didn't understand.

I get the idea of mechanically putting it everywhere so that you can have a warning (or even error) when docs on public methods are omitted. I'm not sure that I agree with it... Source code becomes verbose and cluttered with useless docs (I'm a terseness fiend I guess), with very little added value. It's OK to sometimes rely on programmers to add docs where it makes sense rather than to impose it via warnings/errors.

Re EF Core, note that their "public documentation" is very frequently a note saying that the API in question isn't really public (meant for providers/infrastructure and may change).

Reply:
@austindrencki, let's add comments that it's an infrastructure class. Same in other places.